### PR TITLE
salmon: update version

### PR DESCRIPTION
**Pull request by pabloaledo**
update salmon version 1.10.2

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256. Deferring to @snehring for merge decision.

File: var/spack/repos/builtin/packages/salmon/package.py [hunk 1]
```diff
@@ -55,10 +56,18 @@ class Salmon(CMakePackage):
     depends_on("libgff")
     depends_on("pkgconfig")
     depends_on("curl", when="@0.14.1:")
+    depends_on("htslib", when="@1.10.2")
```
Comment (snehring) [inline var/spack/repos/builtin/packages/salmon/package.py:59]: I'm assuming this'll be a requirement going forward, so probably more appropriate for this to be when="@1.10.2:"

File: var/spack/repos/builtin/packages/salmon/package.py [hunk 2]
```diff
@@ -55,10 +56,18 @@ class Salmon(CMakePackage):
...
+
+    patch("fix_hts.patch", when="@1.10.2")
```
Comment (snehring) [inline var/spack/repos/builtin/packages/salmon/package.py:61]: Did you report this upstream or are they already aware of it separately and plan to fix it in the next release?

**Review by snehring — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/salmon/package.py (hunk 2 above)
Comment (pabloaledo) [reply var/spack/repos/builtin/packages/salmon/package.py:61]: https://github.com/COMBINE-lab/salmon/issues/866

**Review by snehring — APPROVED** [review]
As is this is probably fine for this version. If the issues with cmake and hts get sorted out upstream we won't need these changes to apply to later versions.

**Pull request merged**